pipeline/plot/unit_characteristic_plot.py

```diff
@@ -155,148 +168,202 @@ def plot_unit_bilateral_photostim_effect(probe_insert_key, axs=None):
-    # ipsi_trials = psth.TrialCondition & {'condition_id': 1 if unit_hemi == 'left' else 0}
-    # contra_trials = psth.TrialCondition & {'condition_id': 0 if unit_hemi == 'left' else 1}
-    # ipsi_trials = psth.TrialCondition & {'condition_id': 1 if unit_hemi == 'left' else 0}
-
-    if unit_hemi == 'left':
-        ipsi_trials = psth.Condition.audio_delay_ipsi_hit_nostim()
-        contra_trials = psth.Condition.audio_delay_contra_hit_nostim()
-    else:
-        ipsi_trials = psth.Condition.audio_delay_contra_hit_nostim()
-        contra_trials = psth.Condition.audio_delay_ipsi_hit_nostim()
-
-    ipsi_sel_units = (ephys.Unit * psth.UnitSelectivity
-                      & 'unit_selectivity = "ipsi-selective"').fetch('KEY')
-
-    contra_sel_units = (ephys.Unit * psth.UnitSelectivity
-                        & 'unit_selectivity = "contra-selective"').fetch('KEY')
 
     if axs is None:
         fig, axs = plt.subplots(1, 2, figsize=(20, 20))
     assert axs.size == 2
 
-    period_starts = (experiment.Period & 'period in ("sample", "delay", "response")').fetch('period_start')
+    period_starts = (experiment.Period
+                     & 'period in ("sample", "delay", "response")').fetch(
+                         'period_start')
 
-    # contra-selective units
-    # _plot_stacked_psth_diff(
-    #     (psth.UnitPsth * contra_sel_units & contra_trials).fetch(order_by='unit_posy desc'),
-    #     (psth.UnitPsth * contra_sel_units & ipsi_trials).fetch(order_by='unit_posy desc'),
-    #     ax=axs[0], vlines=period_starts)
+    good_unit = ephys.Unit & {'unit_quality': 'good'}
+    conds_i = (psth.TrialCondition
+               & {'trial_condition_desc':
+                  'good_noearlylick_left_hit'}).fetch('KEY')
 
-    # contra-selective units
+    conds_c = (psth.TrialCondition
+               & {'trial_condition_desc':
+                  'good_noearlylick_right_hit'}).fetch('KEY')
 
-    # compute psths WIP / stalled - see `rework Condition` in psth.py
-    psth_c_c = [psth.UnitPsth.compute_unit_trial_psth(u, contra_trials)
-                for u in contra_sel_units]
-    psth_c_i = [psth.UnitPsth.compute_unit_trial_psth(u, ipsi_trials)
-                for u in contra_sel_units]
+    sel_i = (ephys.Unit * psth.UnitSelectivity
+             & 'unit_selectivity = "ipsi-selective"' & probe_insert_key)
 
-    _plot_stacked_psth_diff(psth_c_c, psth_c_i, ax=axs[0], vlines=period_starts)
-    axs[0].set_title('Contra-selective Units')
+    sel_c = (ephys.Unit * psth.UnitSelectivity
+             & 'unit_selectivity = "contra-selective"' & probe_insert_key)
 
-    # ipsi-selective units
-    # _plot_stacked_psth_diff(
-    #     (psth.UnitPsth * ipsi_sel_units & ipsi_trials).fetch(order_by='unit_posy desc'),
-    #     (psth.UnitPsth * ipsi_sel_units & contra_trials).fetch(order_by='unit_posy desc'),
-    #     ax=axs[1], vlines=period_starts)
+    psth_is_it = (((psth.UnitPsth & conds_i)
```

Comment:
Later on, please add comments to clarify what these are: `PSTH over IPSI trials for IPSI selective units`

Reply:
fixed